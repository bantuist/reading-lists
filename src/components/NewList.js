import React, { Component } from 'react';

// import './Lists/Lists.css';

class NewList extends Component {
  handleSubmit(event) {
    event.preventDefault();
    console.log(this);
    const name = this.input.value;

    this.props.onAddList(name);
    this.input.value = '';
  }
  render() {
    return (
      <div className="new-list-container">
        <form className="new-list-form" onSubmit={(event) => this.handleSubmit(event)}>
          <input
            className="new-list-input"
            ref={(input) => this.input = input}
            type="text"
            placeholder="Create a new list"
          />
        </form>
      </div>
    );
  }
}

export default NewList;
