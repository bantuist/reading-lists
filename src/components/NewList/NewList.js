import React from 'react';

function NewList(props) {
  function addList(event) {
    event.preventDefault();
    const name = inputRef.value;
    props.onAddList(name);
    inputRef.value = '';
  }

  let inputRef;

  return (
    <div className="new-list-container">
      <form className="new-list-form" onSubmit={(event) => addList(event)}>
        <input
          className="new-list-input"
          ref={(input) => inputRef = input}
          type="text"
          placeholder="Create a new list"
        />
      </form>
    </div>
  );
}

export default NewList;
