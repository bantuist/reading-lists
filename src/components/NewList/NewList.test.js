import React from 'react';
import ReactDOM from 'react-dom';
import NewList from './NewList';
import renderer from 'react-test-renderer';

it('should render NewList', () => {
  const tree = renderer.create(<NewList />).toJSON();
  expect(tree).toMatchSnapshot();
});