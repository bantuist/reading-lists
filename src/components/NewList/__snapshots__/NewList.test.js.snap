// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewList 1`] = `
<div
  className="new-list-container"
>
  <form
    className="new-list-form"
    onSubmit={[Function]}
  >
    <input
      className="new-list-input"
      placeholder="Create a new list"
      type="text"
    />
  </form>
</div>
`;
